WordPress/Classes/Services/PostService.m

```diff
@@ -206,7 +221,7 @@ - (void)syncPostsOfType:(NSString *)postType
                    success:^(NSArray *posts) {
                        BOOL hasMore = ([posts count] < PostServiceNumberToFetch) ? NO : YES;
                        [self.managedObjectContext performBlock:^{
-                           [self mergePosts:posts ofType:postType withStatuses:postStatus forBlog:blog purgeExisting:YES completionHandler:^{
+                           [self mergePosts:posts ofType:postType withStatuses:postStatus byAuthor:authorID forBlog:blog purgeExisting:YES completionHandler:^{
```

Comment:
IMO, it doesn't _create_ problems, it just makes more apparent the old problem of gaps in content. 
The same problem appears if you are in `Everyone`, and you switch to `Only Me`, and quickly switch back before it syncs. It will try to sync for Everyone, but since `isSyncing == YES`, it will abort. Now you have the same situation.

Syncing is hard :disappointed: 

I guess it is better than the alternative, but it makes me a bit sad that we keep piling up small syncing bugs and never get around to fix the actual problem.